python/python/lance/dataset.py

```diff
@@ -689,6 +689,83 @@ def cleanup_old_versions(
             td_to_micros(older_than), delete_unverified
         )
 
+    def create_scalar_index(
+        self,
+        column: Union[str, List[str]],
+        index_type: str,
```

Comment:
I'd generally prefer to use `Literal["BTREE"]` for now, if possible. This given better auto-completion options.

Reply:
Or an enum?

Reply:
I ended up going with an enum.  I hadn't really used enums for python before but now that python has better autocomplete I think this might be useful.  Although, `Literal["BTREE"]` should functionally be an enum.  @wjones127 any opinion on which to prefer?

Reply:
I think it's more typical Python style to use strings. It avoids the need to have to import the enum type in order to pass the value.

Reply:
Having tried out both of them now I think I will stick with strings.  Besides the issues that Will mentioned this will also give us consistency between the two create index APIs and between the rust API and the python API.